platform/contracts/admin/src/endpoints.rs

```diff
@@ -273,6 +276,24 @@ fn register_protocol(
     state_contracts::add_protocol(storage, name, protocol).map(|()| response::empty_response())
 }
 
+fn deregister_protocol(
+    storage: &mut dyn Storage,
+    sender: &Addr,
+    migration_spec: ProtocolContracts<MigrationSpec>,
+) -> ContractResult<CwResponse> {
+    for name in state_contracts::protocols(storage)? {
+        let protocol = state_contracts::load_protocol(storage, name.clone())?.contracts;
+
+        if protocol.leaser == sender {
+            return Ok(response::response_only_messages(
+                protocol.migrate_standalone(migration_spec),
+            ));
+        }
+    }
```

Comment:
I would suggest you refactor this in a functional style, a.k.a `findmap`
After all most of the other code, at least the recently modified, follows that style

Reply:
That's a good idea. At first I searched for a `try_find` kind of method but I might have seen it in third-party packages extending iterators. I didn't think of `find_map`. Thanks!